tools/perf/scripts/bench_run_log.py

```diff
@@ -237,7 +243,16 @@ def single_run(self):
         # Log the payload $step_count times
         with Timer() as timer:
             for _ in range(self.num_steps):
-                run.log(payload)
+                if not self.is_unique_payload:
+                    run.log(payload)
+                else:
+                    run.log(PayloadGenerator(
+                        self.data_type, self.num_metrics, self.metric_key_size
+                    ).generate())
```

Comment:
now you are counting the time to generate payload in the log time no?
(you could generate the payload before the call to the log function, also what is the point of using the same payload in terms of perf evaluation?

Reply:
This one was a bit of a catch 22.
Before this change, the test generates the set of metrics (say 100) once and keep logging them.  The goal was to avoid generating too many metrics, and also to avoid counting the CPU and time spent in the payload generation itself which isn't what I want to measure in my benchmark. The benchmark I did was done that way to not count the payload generation.
Other than the benchmark scenario, now I am testing a stress scenario to see if the SDK and our cloud backend can handle 1M metrics.  I don't mind counting the CPU and time for generating the payload because the goal of the stress test was to see if it could finish the run without data loss.  I won't "benchmark" the metrics in the stress test scenario.
And you are right, a better option is to generate the 1M metrics payloads outside and just let the log() step read them.  1M metrics in my case account for 20MB of payload which is still very small in memory usage.  That would keep the metrics overall mch cleaner.  I will do that instead.